esrally/metrics.py

```diff
@@ -254,8 +254,15 @@ def results_template(self):
     def _read(self, template_name):
         with open("%s/resources/%s.json" % (self.script_dir, template_name), encoding="utf-8") as f:
             template = json.load(f)
-            template["settings"]["index"]["number_of_shards"] = self._number_of_shards
-            template["settings"]["index"]["number_of_replicas"] = self._number_of_replicas
+            if self._number_of_shards is not None:
+                if int(self._number_of_shards) < 1:
+                    raise exceptions.SystemSetupError(
+                        f"The setting: datastore.number_of_shards must be >= 1. Please "
+                        f"check the configuration in {config.ConfigFile().location}/rally.ini"
```

Comment:
This won't work, it will instantiate a new `ConfigFile()` object with `None` as `config_name` (see https://github.com/elastic/rally/blob/f4302fbbc38bf4e7c00cd43ac9b363f26e579196/esrally/config.py#L43). Therefore it'll always return `rally.ini`.

We need to use `self._config.config_file.location` as mentioned in the [previous comment](https://github.com/elastic/rally/pull/1312#discussion_r689351906).